### ISSUE TITLE
Variable substitution in constraint

### ISSUE THREAD

**Issue by gittycat** [docker/compose#3188]
Variable extrapolation doesn't occur inside a constraint environment variable.

```
version: "2"
services:
  app:
    image: alpine:3.3
    environment:
      "constraint:node==${VM_PREFIX}node01":
    command: echo 'hello'
```
`docker-compose up -d` gives the following error:
```
Creating swarm_app_1
ERROR: Value '${VM_PREFIX}node01' is invalid
```

docker: 1.10.3
docker-compose: 1.6.2
virtualbox, swarm, OSX CLI

**Comment by dnephin** [comment]
I don't think this is correct. It should be:

```yaml
environment:
  "constraint:node" : "=${VM_PREFIX}node01"
```

which I think should work.

**Comment by gittycat** [comment]
Yes, that works. I would have not guessed that syntax though.

The syntax used most often for an environment is 

I first tried
```
environment:
  constraint: "node==somenode"
```
but that didn't work so I then used 
```
environment:
  "constraint:node==somenode":
```
**which does work** at constraining by node but variable substitution brings in the error as I mentioned.

I suggest you include the syntax you list in your documentation along with the other variant
```
environment:
  - "contraint:node==somenode"
```
Thanks
